src/selectors/functions/builtInFunctions.numeric.js

```diff
@@ -115,18 +115,18 @@ function fnRound (halfToEven, _dynamicContext, sequence, precision) {
 	}
 }
 
-function fnNumber (_dynamicContext, sequence) {
+function fnNumber (dynamicContext, arg = undefined) {
+	const sequence = arg ? arg : dynamicContext.contextItem;
+
 	if (sequence.isEmpty()) {
 		return Sequence.singleton(createAtomicValue(NaN, 'xs:double'));
 	}
+
 	try {
 		return Sequence.singleton(castToType(sequence.first(), 'xs:double'));
```

Comment:
I'd say it would be better to directly use tryCastToType, and check for errors here.